src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -83,6 +83,18 @@ public TimeSpan UserProcessorTime
             }
         }
 
+        partial void EnsureHandleCountPopulated()
+        {
+            if (_processInfo.HandleCount <= 0 && _haveProcessId)
+            {
+                string path = "/proc/" + _processId.ToString() + "/fd";
+                if (Directory.Exists(path))
```

Comment:
This is insufficient to deal with the race condition.  The moment after this check, the process could go away.  You need to catch the exception.